code/game/objects/items/candle.dm

```diff
@@ -8,80 +8,70 @@
 
 	var/wax = 200
 	var/lit = 0
-	proc
-		light(var/flavor_text = "\red [usr] lights the [name].")
+	var/flavor_text
 
+/obj/item/candle/update_icon()
+	var/i
+	if(wax > 150)
+		i = 1
+	else if(wax > 80)
+		i = 2
+	else i = 3
+	icon_state = "candle[i][lit ? "_lit" : ""]"
 
-	update_icon()
-		var/i
-		if(wax>150)
-			i = 1
-		else if(wax>80)
-			i = 2
-		else i = 3
-		icon_state = "candle[i][lit ? "_lit" : ""]"
-
-
-	attackby(obj/item/weapon/W as obj, mob/user as mob)
-		..()
-		if(istype(W, /obj/item/weapon/weldingtool))
-			var/obj/item/weapon/weldingtool/WT = W
-			if(WT.isOn()) //Badasses dont get blinded by lighting their candle with a welding tool
-				light("\red [user] casually lights the [name] with [W], what a badass.")
-		else if(istype(W, /obj/item/weapon/lighter))
-			var/obj/item/weapon/lighter/L = W
-			if(L.lit)
-				light()
-		else if(istype(W, /obj/item/weapon/match))
-			var/obj/item/weapon/match/M = W
-			if(M.lit)
-				light()
-		else if(istype(W, /obj/item/candle))
-			var/obj/item/candle/C = W
-			if(C.lit)
-				light()
+/obj/item/candle/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	..()
+	if(istype(W, /obj/item/weapon/weldingtool))
+		var/obj/item/weapon/weldingtool/WT = W
+		if(WT.isOn()) //Badasses dont get blinded by lighting their candle with a welding tool
+			light("<span class='warning'>[user] casually lights [src] with [W], what a badass.</span>")
+	else if(istype(W, /obj/item/weapon/lighter))

```

Comment:
Except all the messages are the same, and snowflake code is messy. is_hot() is much neater. You could even leave the welder message in, it's just pointless to have to add in exceptions to candle code when a general system is better.